pkg/ds/daemon_set.go

```diff
@@ -563,3 +567,32 @@ func (ds *daemonSet) CurrentPods() (types.PodLocations, error) {
 
 	return result, nil
 }
+
+func (ds *daemonSet) NodesToUpdate(superset types.PodLocations) (types.PodLocations, error) {
+	reality, _, err := ds.kpStore.AllPods(kp.REALITY_TREE)
```

Comment:
I echo the concerns that this will fetch too much data. Perhaps it would be better to fetch reality for individual pods. It would mean more requests, but they could be spread out over time since we don't want to update all nodes at once anyway.

You're more familiar with the code right now than I am, but would it make for sense for this strategy to be implemented inside the replicator instead of being a filter that controls the nodes that the replicator sees?